test/support/utils.rb

```diff
@@ -71,44 +71,50 @@ def collect_debuggee_backlog test_info
 
     MULTITHREADED_TEST = !(%w[1 true].include? ENV['RUBY_DEBUG_TEST_DISABLE_THREADS'])
 
-    # This method will execute both local and remote mode by default.
-    def debug_code(program, boot_options: '-r debug/start', remote: true, &block)
-      write_temp_file(strip_line_num(program))
-      @scenario = []
-      block.call
-      @scenario.freeze
-      inject_lib_to_load_path
-
-      ENV['RUBY_DEBUG_NO_COLOR'] = 'true'
-      ENV['RUBY_DEBUG_TEST_MODE'] = 'true'
-      ENV['RUBY_DEBUG_NO_RELINE'] = 'true'
+    def debug_code(program, ruby: nil, rdbg: nil, remote: true, &test_steps)
+      if ruby
+        run_ruby(program, options: ruby, &test_steps)
+      else
+        run_rdbg(program, remote: remote, &test_steps)
```

Comment:
I will support the `options` argument when adding tests for `rdbg` command, which will be the next PR after this is merged.